scripts/street_sign_recognizer.py

```diff
@@ -17,45 +19,139 @@ def __init__(self):
+        cv2.namedWindow('threshold_image')
+        self.hsv_lb = np.array([22, 134, 166]) # hsv lower bound
+        cv2.createTrackbar('H lb', 'threshold_image', self.hsv_lb[0], 255, self.set_h_lb)
+        cv2.createTrackbar('S lb', 'threshold_image', self.hsv_lb[1], 255, self.set_s_lb)
+        cv2.createTrackbar('V lb', 'threshold_image', self.hsv_lb[2], 255, self.set_v_lb)
+        self.hsv_ub = np.array([44, 255, 255]) # hsv upper bound
+        cv2.createTrackbar('H ub', 'threshold_image', self.hsv_ub[0], 255, self.set_h_ub)
+        cv2.createTrackbar('S ub', 'threshold_image', self.hsv_ub[1], 255, self.set_s_ub)
+        cv2.createTrackbar('V ub', 'threshold_image', self.hsv_ub[2], 255, self.set_v_ub)
+
+        images = {
+        "left": '../images/leftturn_box_small.png',
+        "right": '../images/rightturn_box_small.png',
+        "uturn": '../images/uturn_box_small.png'
+        }
+
+        self.tm = TemplateMatcher(images)
+
+    def set_h_lb(self, val):
+        """ set hue lower bound """
+        self.hsv_lb[0] = val
+
+    def set_s_lb(self, val):
+        """ set saturation lower bound """
+        self.hsv_lb[1] = val
+
+    def set_v_lb(self, val):
+        """ set value lower bound """
+        self.hsv_lb[2] = val
+
+    def set_h_ub(self, val):
+        """ set hue upper bound """
+        self.hsv_ub[0] = val
+
+    def set_s_ub(self, val):
+        """ set saturation upper bound """
+        self.hsv_ub[1] = val
+
+    def set_v_ub(self, val):
+        """ set value upper bound """
+        self.hsv_ub[2] = val
 
     def process_image(self, msg):
         """ Process image messages from ROS and stash them in an attribute
             called cv_image for subsequent processing """
         self.cv_image = self.bridge.imgmsg_to_cv2(msg, desired_encoding="bgr8")
+        self.hsv_image = cv2.cvtColor(self.cv_image, cv2.COLOR_BGR2HSV)
+        self.threshold_image = cv2.inRange(self.hsv_image, self.hsv_lb, self.hsv_ub)
 
         left_top, right_bottom = self.sign_bounding_box()
+        #print left_top, right_bottom
         left, top = left_top
         right, bottom = right_bottom
 
+        # draw bounding box rectangle
+        cv2.rectangle(self.hsv_image, left_top, right_bottom, color=(255, 0, 0), thickness=5)
+
         # crop bounding box region of interest
         cropped_sign = self.cv_image[top:bottom, left:right]
+        grayscale_sign = cv2.cvtColor(cropped_sign, cv2.COLOR_BGR2GRAY)
+        print self.tm.predict(grayscale_sign)
 
-        # draw bounding box rectangle
-        cv2.rectangle(self.cv_image, left_top, right_bottom, color=(0, 0, 255), thickness=5)
+        
 
     def sign_bounding_box(self):
         """
         Returns
         -------
         (left_top, right_bottom) where left_top and right_bottom are tuples of (x_pixel, y_pixel)
             defining topleft and bottomright corners of the bounding box
+
+            I found this to be a good resource on contours
```

Comment:
I really like comments with links to resources, since it's both honest (in terms of where your code was inspired) but also helpful for a reviewer like me to learn some of your process in choosing this algorithm).